### PR TITLE
Deploy jars to Maven central

### PR DESCRIPTION
**Pull request by nitay**
Deployed to Maven Central [here](https://repo1.maven.org/maven2/co/actioniq/vault-utilities/) using Sonatype

This makes it easier to get at this library, no need to any special S3 resolving
Note that using Sonatype/Central ensures that we always publish javadoc & source jars as it won't allow you to publish otherwise.

### CODE REVIEW

**Comment by MasterDDT** [comment]
Do you know how to fix this warning we get starting up almost any app

```
                     SLF4J: Found binding in [jar:file:/home/jenkins/workspace/ActionIQ_aiq_PR-36039/aiq/.pants.d/resolve/coursier/e2197a99ff41/cache/relative/s3/s3.amazonaws.com/aiq-artifacts/releases/net/wedjaa/ansible/vault/vault-utilities/1.2.0/vault-utilities-1.2.0.jar!/org/slf4j/impl/StaticLoggerBinder.class]
```

I think wiggles said something about we need to rebuild this thing without the slf4j libraries, so it uses our own? Nothing breaks with logging tho

File: pom.xml [hunk 1]
```diff
@@ -57,29 +80,109 @@
...
+    </distributionManagement>
+    <profiles>
+        <profile>
+            <id>ossrh</id>
+            <activation>
+                <activeByDefault>true</activeByDefault>
+            </activation>
+            <properties>
+                <gpg.executable>gpg2</gpg.executable>
+                <gpg.passphrase>the_pass_phrase</gpg.passphrase>
```
Comment (MasterDDT) [inline pom.xml]: Is this like a secret?

Comment (nitay) [reply pom.xml]: oh good catch this shouldnt be needed I'll remove it

File: dependency-reduced-pom.xml [hunk 2]
```diff
@@ -1,10 +1,35 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
   <modelVersion>4.0.0</modelVersion>
-  <groupId>net.wedjaa.ansible.vault</groupId>
+  <groupId>co.actioniq</groupId>
   <artifactId>vault-utilities</artifactId>
-  <version>1.2.0</version>
+  <name>vault-utilities</name>
+  <version>1.2.1</version>
```
Comment (MasterDDT) [inline dependency-reduced-pom.xml]: Did you publish 1.2.1? So then this should be `1.2.2-SNAPSHOT` to match below?

**Review by MasterDDT — APPROVED** [review]

**Comment by nitay** [comment]
@MasterDDT oh interesting you're right it pulls in the logger impls, I'll remoe those just leave the API

File: dependency-reduced-pom.xml (hunk 2 above)
Comment (nitay) [reply dependency-reduced-pom.xml]: I don't think this file needs to even be checked in gonna remove it and git ignore it

**Pull request merged**